### PR TITLE
Implement ContextHandle inside GGeom directly

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
@TeXitoi Does the API looks better this way? If so, we'll have to add a lot more examples in the docs stating that setting `GContextHandle` inside an object is better than just calling the object.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
After discussion, I'll change the following things:

 * No more usage of functions which not use `HandleContext` (the ones with name which doesn't end with "_r") because they're not thread-safe.
 * All objects will now have a `HandleContext` (in the object or given when calling the method, not clear yet, to be tested).
 * Save X last errors/notifications so the user can take a look at them whenever they want.

File: src/context_handle.rs [hunk 1]
```diff
@@ -31,14 +74,36 @@ impl GContextHandle {
         initialize();
```
Comment (TeXitoi) [inline src/context_handle.rs]: initialize is not needed as it doesn't have `_t`

Comment (GuillaumeGomez) [reply src/context_handle.rs]: Good point. I'll remove it everywhere.

File: src/coord_seq.rs [hunk 2]
```diff
@@ -1,105 +1,254 @@
...
+    ///                               .expect("failed to create CoordSeq");
+    /// let positions2: &[(f64, f64)] = &[(0., 0.), (1., 2.), (14., 5.)];
+    /// for (pos, (x, y)) in positions2.into_iter().enumerate() {
+    ///     coord_seq2.set_x(pos, *x).expect("failed to set x...");
+    ///     coord_seq2.set_y(pos, *y).expect("failed to set y...");
+    /// }
+    /// assert!(coord_seq2.get_x(1) == Ok(1.));
+    /// ```
+    pub fn new(size: u32, dims: CoordDimensions) -> GResult<CoordSeq<'a>> {
         initialize();
```
Comment (TeXitoi) [inline src/coord_seq.rs]: I say nooo, nooo, nooo, nooo ;-)

**Comment by GuillaumeGomez** [comment]
Well, that's why I was against this but the user can wrap it pretty easily by using channels.

**Comment by GuillaumeGomez** [comment]
I removed `initialize` calls.

**Comment by GuillaumeGomez** [comment]
Beta version of rust has memory leaks on threads apparently. That's pretty funny. :3

**Pull request merged**